configs/components/facter.rb

```diff
@@ -61,7 +61,7 @@
   java_home = ''
   java_includedir = ''
   case platform.name
-  when /(el-(6|7)|fedora-f24)/
```

Comment:
Did we just forget to add fedora 25 for this case? Should we have a java build requires for fc25?

Reply:
or should we have a generic fedora entry until there's a version it breaks for?

Reply:
@McdonaldSeanp @branan ^^

Reply:
We need the java build-requires for Master platforms so that libfacter.so has JNI support.

Reply:
It looks like we build with JNI support on a ton of platforms below - I'm not certain what we're shipping puppetserver on these days, but we should probably clean this up so we're not building against Java when we don't need to.

Reply:
That all seems like work separate from this PR though.

Reply:
For sure. I don't think there's anything we need to do as part of this PR